xtrack/line.py

```diff
@@ -726,101 +768,177 @@ def get_s_position(self, at_elements=None, mode="upstream"):
             return s
 
     def remove_markers(self, inplace=True, keep=None):
-        if not inplace:
-            raise NotImplementedError
+
         self._frozen_check()
 
-        if isinstance(keep, str):
+        if keep is None:
+            keep = []
+        elif isinstance(keep, str):
             keep = [keep]
 
-        names = []
+        newline = Line(elements=[], element_names=[])
+
         for ee, nn in zip(self.elements, self.element_names):
-            if isinstance(ee, Marker):
-                if keep is None or nn not in keep:
-                    continue
-            names.append(nn)
+            if isinstance(ee, Marker) and nn not in keep:
+                continue
+            newline.append_element(ee, nn)
 
-        self.element_names = names
-        return self
+        if inplace:
+            self.element_names = newline.element_names
+            self.element_dict  = newline.element_dict
```

Comment:
I think rebinding element_dict breaks the xdeps expressions

Reply:
xdeps updates values in `_owner` attribute of the ref. It is possible to replace it to a new dictionary if needed.

Reply:
I reinitialised the ref; that should do it I believe.